cooker/lib/pages.js

```diff
@@ -241,58 +392,27 @@ async function pages(file, localeFile) {
 
   logger.start("Generating pages");
 
-  // go throught all of them
+  // go through all of them
   let promises = [];
 
   for (let file of pugFiles) {
     // get the file information and locale files
     let templateInfo = path.parse(file);
 
-    // The data file for MD. This just includes any object data within an MD file, if that's what we're compiling
-    const mdData = {};
+    // Container variable for markdown data
+    let mdData;
 
-    // Otherwise compile as a pug file
     // if file starts with underscore, we ignore it, expected as standard 👍
     if (templateInfo.name.charAt(0) == "_") continue;
 
     // If we're a markdown file, compile with grey-matter
+    // This is a great pattern that we can just extend as we need to to render different kinds of files.
     if (templateInfo.ext.toLowerCase() === ".md") {
-      const mdfile = matter.read(file);
-      const name = templateInfo.name;
-      const dir = templateInfo.dir;
-      const templateFile = path.join(templateInfo.dir, mdfile.data.template);
-
-      // If we have a template file defined in the grey-matter file
-      if (mdfile.data.template) {
-        // If the template doesn't exist, warn and continuer
-        if (!fse.pathExistsSync(templateFile)) {
-          logger.warning([
-            `Error compiling ${path.relative(
-              config.CWD,
-              file
-            )}. The markdown file calls for the template ${path.relative(
-              config.CWD,
-              templateFile
-            )} which does not exist`,
-          ]);
-          continue;
-        } else {
-          // Fill out the markdown data content
-          // Convert the content using markdown.
-          mdData.content = Markdown.render(mdfile.content);
-          // Transcribe the returned data components to the markdown data object
-          for (let t in mdfile.data) {
-            mdData[t] = mdfile.data[t];
-          }
-          // Parse the template file
-          templateInfo = path.parse(templateFile);
-          // ... But set the name of the output to the name of the md file
-          templateInfo.name = name;
-          // ... and set the dir to the md file dir
-          templateInfo.dir = dir;
-          // Finally update the file to parse to the provided template file.
-          file = templateFile;
-        }
+      try {
+        ({ templateInfo, file, mdData } = parseMarkdown(file, templateInfo));
```

Comment:
Small lil thing. I think there is a set of extra parenthesis in here.

Reply:
Yeah, prettier was complaining about the naked destructure, but I think that's just prettier being basically configured.

Reply:
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#assignment_without_declaration

Reply:
I noticed what was going on with the assignment, was just curious about the parenthesis, but makes sense that prettier didn't like that.